simple_metering.cpp

```diff
@@ -0,0 +1,200 @@
+                        configUpdated = true;
+                    }
+                }
+                sensor->setZclValue(updateType, ind.srcEndpoint(), METERING_CLUSTER_ID, attrId, attr.numericValue());
+            }
+                break;
+
+            case 0x0302: // Interface Mode
+            {
+                if (zclFrame.manufacturerCode() == VENDOR_DEVELCO && sensor->modelId() == QLatin1String("ZHEMI101"))
+                {
+                    quint16 interfaceMode = attr.numericValue().u16;
+                    quint8 mode = 0;
+                    
+                    if (interfaceMode == 0x0000) { mode = 1; }          // Pulse Counting on an Electricity Meter – Unit KWh
+                    else if (interfaceMode == 0x0001) { mode = 2; }     // Pulse Counting on a Gas Meter – Unit m3
+                    else if (interfaceMode == 0x0002) { mode = 3; }     // Pulse Counting on a Water Meter – Unit m3
+                    else if (interfaceMode == 0x0100) { mode = 4; }     // Kamstrup KMP Protocol
+                    else if (interfaceMode == 0x0101) { mode = 5; }     // Not Supported - Linky Protocol
+                    else if (interfaceMode == 0x0102) { mode = 6; }     // DLMS-COSEM - IEC62056-21 mod A
+                    else if (interfaceMode == 0x0103) { mode = 7; }     // P1 Dutch Standard – DSMR 2.3 Version
+                    else if (interfaceMode == 0x0104) { mode = 8; }     // P1 Dutch Standard – DSMR 4.0 Version
+                    
+                    item = sensor->item(RConfigInterfaceMode);
+                    if (item && item->toNumber() != mode && mode > 0 && mode < 9)
+                    {
+                        item->setValue(mode);
+                        enqueueEvent(Event(RSensors, RConfigInterfaceMode, sensor->id(), item));
+                        configUpdated = true;
+                    }
+                }
+                sensor->setZclValue(updateType, ind.srcEndpoint(), METERING_CLUSTER_ID, attrId, attr.numericValue());
+            }
+                break;
+
+            default:
+                break;
+            }
+        }
+
+        if (stateUpdated)
+        {
+            sensor->updateStateTimestamp();
+            enqueueEvent(Event(RSensors, RStateLastUpdated, sensor->id()));
+        }
+
+        if (configUpdated || stateUpdated)
+        {
+            updateSensorEtag(&*sensor);
+            sensor->setNeedSaveDatabase(true);
+            queSaveDb(DB_SENSORS, DB_SHORT_SAVE_DELAY);
+        }
+    }
+}
+
+/*! Write Attribute on thermostat cluster.
+ *  Iterate over every day and get schedule for each day.
+   \param task - the task item
+   \param attrId
+   \param attrType
+   \param attrValue
+   \return true - on success
+           false - on error
+ */
+bool DeRestPluginPrivate::addTaskSimpleMeteringReadWriteAttribute(TaskItem &task, uint8_t readOrWriteCmd, uint16_t attrId, uint8_t attrType, uint32_t attrValue, uint16_t mfrCode)
```

Comment:
Nothing wrong, but I'd suggest to stick to quint8, 16, ... instead uint8_t types for consistency.

Reply:
Got the idea. However, if you're fine with it, I'd prefer to leave it for now as it is. Since we got this all over the code, I'd rather got with a new PR to address this consistently. What do you think?

Reply:
Yes fine by me :)